sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -118,7 +120,25 @@ public static PipelineOptions createPipelineOptions(Map<String, String> map) {
     for (Map.Entry<String, String> entry : map.entrySet()) {
       args[i++] = "--" + entry.getKey() + "=" + entry.getValue();
     }
-    PipelineOptions options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+
+    PipelineOptions options = null;
+    try {
+      options = PipelineOptionsFactory.fromArgs(args).withValidation().create();
+    } catch (IllegalArgumentException e) {
+      Matcher matcher =
+          Pattern.compile("Class (.+) missing a property named \\'(.+)\\'\\.(.*)")
+              .matcher(e.getMessage());
+      if (matcher.matches()) {
+        throw new IllegalArgumentException(
+            String.format(
+                "You may have SET an unregistered option '%s', which could have failed current "
+                    + "query. You could run 'RESET %s' in Shell to reset the option.",
+                matcher.group(2), matcher.group(2)));
```

Comment:
Would be nice if the original exception was preserved. Can you add it as an argument to the new one?